projects/packages/videopress/src/client/admin/components/video-row/index.tsx

```diff
@@ -0,0 +1,263 @@
+								: __( 'Public', 'jetpack-videopress-pkg' ) }
+						</span>
+					</>
+				) }
+			</div>
+			<div className={ styles.duration }>
+				<span>{ isSmall && durationLabel }</span>
+				<span>{ duration }</span>
+			</div>
+			{ Number.isFinite( plays ) && (
+				<div className={ styles.plays }>
+					<span>{ isSmall && playsLabel }</span>
+					<span>{ plays }</span>
+				</div>
+			) }
+			{ ! isSmall && <div className={ styles.upload }>{ uploadDate }</div> }
+		</div>
+	);
+};
+
+const VideoRow = ( {
+	videoTitle,
+	posterImage,
+	duration,
+	uploadDate,
+	plays = null,
+	isPrivate = false,
+	onClickEdit,
+}: {
+	videoTitle: string;
+	posterImage: string;
+	duration: number;
+	uploadDate: string;
+	plays: number;
+	isPrivate: boolean;
+	onClickEdit?: () => void;
+} ) => {
+	const [ isSmall ] = useBreakpointMatch( 'sm' );
+	const [ showActions, setShowActions ] = useState( false );
+	const [ keyPressed, setKeyDown ] = useState( false );
+	const [ expanded, setExpanded ] = useState( false );
+	const textRef = useRef( null );
+
+	const durationInMinutesAndSeconds = millisecondsToMinutesAndSeconds( duration );
+	const uploadDateFormatted = dateI18n( 'M j, Y', uploadDate, null );
+	const isSpaceOrEnter = code => code === 'Space' || code === 'Enter';
+	const isEllipsisActive = textRef?.current?.offsetWidth < textRef?.current?.scrollWidth;
+	const showTitleLabel = ! isSmall && isEllipsisActive;
+	const showStats = ( ! showActions && ! isSmall ) || ( isSmall && expanded );
+	const showBottomButton = isSmall && expanded;
+
+	const editVideoLabel = __( 'Edit video details', 'jetpack-videopress-pkg' );
+	const wrapperAriaLabel = sprintf(
+		/* translators: 1 Video title, 2 Video duration, 3 Video upload date */
+		__(
+			'Video: %1$s, Duration: %2$s, Upload Date: %3$s. Click to edit details.',
+			'jetpack-videopress-pkg'
+		),
+		videoTitle,
+		durationInMinutesAndSeconds,
+		uploadDateFormatted
+	);
+
+	const editDetailsButton = (
+		<Button size="small" onClick={ onClickEdit } fullWidth={ isSmall }>
+			{ editVideoLabel }
+		</Button>
+	);
+
+	const toggleExpand = () => {
+		setExpanded( current => ! current );
+	};
+
+	const handleKeyDown = e => {
+		if ( isSpaceOrEnter( e?.code ) ) {
+			setKeyDown( true );
+		}
+	};
+
+	const handleKeyUp = e => {
+		if ( isSpaceOrEnter( e?.code ) ) {
+			setKeyDown( false );
+		}
+	};
+
+	const handleOver = () => {
+		setShowActions( true );
+	};
+
+	const handleLeave = () => {
+		setShowActions( false );
+	};
+
+	const handleClick = () => {
+		toggleExpand();
+	};
+
+	const wrapperActions = {
+		onKeyDown: isSmall ? null : handleKeyDown,
+		onKeyUp: isSmall ? null : handleKeyUp,
+		onMouseOver: isSmall ? null : handleOver,
+		onMouseLeave: isSmall ? null : handleLeave,
+		'aria-label': wrapperAriaLabel,
+	};
+
+	// If small screen, expand on click at info wrapper
+	const infoWrapperActions = { onClick: isSmall ? handleClick : null };
+
+	return (
+		<div
+			{ ...wrapperActions }
```

Comment:
I don't see the need to create an object and then destructuring it.

Reply:
Done https://github.com/Automattic/jetpack/pull/25798/commits/838359bad832b622eaea67e2a4da7fcacd0bdd27